lib/elixir/lib/enum.ex

```diff
@@ -717,30 +717,64 @@ defmodule Enum do
   """
   @spec fetch(t, integer) :: {:ok, element} | :error
   def fetch(enumerable, index)
+  def fetch([], index) when is_integer(index),
```

Comment:
Generally speaking, I don't like those clauses for lists. The reason is that, if we implement `fetch` for lists altogether, those cases will be equally as fast, and lists will be faster for *all* cases. So if we want to make lists faster, we should make all cases faster.